### PR TITLE
CDAP-5117 Initializing the ProgramContextAware for distributed mode and also passing the wrapped dataset framework.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-5117
Testing on the cluster in progress..

### CODE REVIEW

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT3673-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java [hunk 1]
```diff
@@ -184,6 +187,15 @@ public WorkflowToken getWorkflowToken() {
     return workflowToken;
   }
 
+  /**
+   * Return the map of local dataset names to the names assigned to the for the current run of the Workflow,
+   * if MapReduce is running inside Workflow. Otherwise {@code null} is returned.
+   */
+  @Nullable
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java]: Why need null? Can't be an empty map?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java [hunk 2]
```diff
@@ -54,10 +54,12 @@
  */
 public final class MapReduceContextConfig {
 
+  public static final String HCONF_ATTR_RUN_ID = "hconf.program.run.id";
+  public static final String HCONF_ATTR_WORKFLOW_LOCAL_DATASET_NAME_MAPPING
+    = "hconf.program.workflow.local.dataset.name.mapping";
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java]: Just call it "hconf.program.dataset.name.mapping" to make it more generic.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java [hunk 3]
```diff
@@ -54,10 +54,12 @@
  */
 public final class MapReduceContextConfig {
 
+  public static final String HCONF_ATTR_RUN_ID = "hconf.program.run.id";
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java]: Why need public?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java [hunk 4]
```diff
@@ -54,10 +54,12 @@
...
+  public static final String HCONF_ATTR_WORKFLOW_LOCAL_DATASET_NAME_MAPPING
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java]: Why public?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java [hunk 5]
```diff
@@ -151,6 +153,13 @@ public ProgramController run(final Program program, ProgramOptions options) {
                                     BasicWorkflowToken.class);
     }
 
+    Map<String, String> workflowLocalDatasetNameMapping = null;
+    if (arguments.hasOption(ProgramOptionConstants.WORKFLOW_LOCAL_DATASET_NAME_MAPPING)) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java]: Do we need this? Can't get it from the `DatasetFramework`? It's really get messy with this map flowing around with serializing and deserializing in different context. If not from the dataset framework, can it be in something more generic?

**Comment by sagarkapare** [comment]
This PR is no longer required. Closing.

**Pull request closed without merging**